Add train station generator

AngusDSR opened the pull request:
### Motivation / Background

Currently, Faker doesn't have a train station generator, and this can be a problem for testing location-based applications or data pipelines that deal with transit data. Having a train station generator would add value to the Faker library and allow for more comprehensive testing of these types of applications.

### Additional information
One alternative to adding a new train station generator would be to manually create a list of station names and use them in tests. However, this can be time-consuming and may not provide enough variety in the generated data.

Another alternative would be to use an external library or API to generate train station names, but this would add an additional dependency to the project and may not integrate well with Faker.

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,26 @@
+        #
+        # @faker.version next
+        def name(type:, region:)
+          fetch("train_station.#{type}.#{region}")
```
I wonder if we should keep the pattern of organizing the locale by region, then type/size, similar to airports. What do you think?

AngusDSR replied:
Yep okay, just saw this. I updated readme and docs to read region, then type... and then I decided to update the yml and order of the code to reflect this too.

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,26 @@
+        # Faker::Travel::TrainStation.name(type: 'metro', region: 'united_kingdom') => "Brockley"
+        #
+        # @faker.version next
+        def name(type:, region:)
```
would it be possible for this generator to support calls without any arguments, or only some of the?

e.g.:
```
 Faker::Travel::TrainStation.name
 Faker::Travel::TrainStation.name(type: 'metro')
```

AngusDSR replied:
Thanks @thdaraujo, managed to find some time to set this functionality up on my latest push.

thdaraujo reviewed (commented):
nice, thanks for working on this! Left a few questions.

AngusDSR commented:
**For the Travel > TrainStations faker:**
- added support for optional argument inputs
- added error handling
- allowed for case insensitivity
- extended testing to related to the above points
- Restructured YAML and code to give precedence to 'region' over 'type'

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,48 @@
+        # @faker.version next
+        def name(region: nil, type: nil)
+          regions = translate('faker.train_station').keys
+          types = translate('faker.train_station.united_kingdom').keys
```
I think we can just limit this option to `['metro', 'railway']` for now:
```suggestion
          types = ['metro', 'railway']
```

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,48 @@
+            type = sample(types)
+          end
+
+          fetch("train_station.#{region.downcase}.#{type.downcase}")
```
it's nice that symbols and mixed-cased strings are easily supported by the generator, but at the same time I don't want people relying on that.

The reason is that if we support `United_Kingdom` from the start, we might end up needing to support `United Kingdom` or `united-kingdom` in the future, and similar things.

I think we should keep it simple for now and only offer support for the simple cases, changing this line to:
```suggestion
          fetch("train_station.#{region}.#{type}")
```

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,48 @@
+          fetch("train_station.#{region.downcase}.#{type.downcase}")
+        end
+
+        def validate_arguments(argument, correct_values, argument_name)
```
maybe this method should be private?

thdaraujo reviewed (commented):
good stuff, thanks for changing the code!

LGTM verall, but I left a few comments. I think we should keep it simple and not downcase the types and regions for now.

AngusDSR commented:
- removed downcasing and symbol compatibility
- created #fill_missing_inputs_with_samples method and moved into private with #validate_arguments

thdaraujo reviewed (approved):
looks great!

thdaraujo commented on lib/faker/travel/train_station.rb:
```diff
@@ -0,0 +1,55 @@
+
+          raise ArgumentError, "'#{argument}' not found, #{argument_name} can be blank, or one of the following, as strings: #{correct_values.join(', ')}"
+        end
+
```
```suggestion
```
